cmd/jcsd/main.go

```
@@ -0,0 +1,79 @@
+}
+
+func handleConnection(conn net.Conn) {
+	defer conn.Close()
+
+	decoder := json.NewDecoder(conn)
+	encoder := json.NewEncoder(conn)
+
+	var req Request
+	if err := decoder.Decode(&req); err != nil {
+		fmt.Fprintln(os.Stderr, "Decode error:", err)
+		encoder.Encode(Response{Error: "invalid JSON"})
```

golangci-lint (1.64.8)

Error return value of `encoder.Encode` is not checked

(errcheck)